images/ceph/toolbox.sh

```diff
@@ -46,19 +41,25 @@ EOF
 
 # watch the endpoints config file and update if the mon endpoints ever change
 watch_endpoints() {
-    # get the timestamp for the target of the soft link
+  # get the timestamp for the target of the soft link
+  real_path=$(realpath ${MON_CONFIG})
+  initial_time=$(stat -c %Z "${real_path}")
+  while true; do
     real_path=$(realpath ${MON_CONFIG})
-    initial_time=$(stat -c %Z ${real_path})
-    while true; do
-       real_path=$(realpath ${MON_CONFIG})
-       latest_time=$(stat -c %Z ${real_path})
-
-       if [[ "${latest_time}" != "${initial_time}" ]]; then
-         write_endpoints
-         initial_time=${latest_time}
-       fi
-       sleep 10
-    done
+    latest_time=$(stat -c %Z "${real_path}")
+    
+    if [[ "${latest_time}" != "${initial_time}" ]]; then
+      write_endpoints
+      initial_time=${latest_time}
+    fi
+    
+    # Enable or disable Ceph cluster debug logs
+    if [ -n "$CEPH_DEBUG_LEVEL" ]; then
+      source enable_disable_ceph_debug_logs.sh "$CEPH_DEBUG_LEVEL"
```

Comment:
I'm still confused by this approach of setting an env var so that the toolbox pod will restart and automatically run the script. This also means the script would be re-run every 10s, just like we update the mon endpoints every 10s in case they're updated. 

The pattern that is more natural to me is:
- Add the script to the rook image, making the script available in the toolbox.
- Connect to the toolbox and run the desired script.

This approach makes it simple to drop in new scripts, have arbitrary arguments, etc. The documentation would also be simple to say they just have to connect to the toolbox and run a script.

Reply:
I think this is a good point. We can just tell users to run `kubectl -n rook-ceph exec -it deploy/rook-ceph-tools -- enable_disable_ceph_debug_logs.sh <level>`

I also personally prefer `-` over `_` for file names to reduce the amount I have to press <Shift>.